### PR TITLE
Ticket32178 Removes trailing spaces in control event log

### PR DESCRIPTION
**Pull request by amadeuszpawlik**
Removes all trailing spaces in control_event_logmsg

### CODE REVIEW

File: src/feature/control/control_events.c [hunk 1]
```diff
@@ -1318,6 +1318,10 @@ control_event_logmsg(int severity, log_domain_mask_t domain, const char *msg)
       for (cp = b; *cp; ++cp)
         if (*cp == '\r' || *cp == '\n')
           *cp = ' ';
+
+      /* Remove trailing spaces */
+      for (--cp; *cp == ' '; --cp)
```
Comment (teor2345) [inline src/feature/control/control_events.c]: Please don't write past the beginning of a string that only has spaces in it:
```suggestion
      for (--cp; *cp == ' ' && cp >= b; --cp)
```

Comment (amadeuszpawlik) [reply src/feature/control/control_events.c]: Agreed;
In such case, should an empty string be sent (send_control_event(...)) at all?

Comment (teor2345) [reply src/feature/control/control_events.c]: That's a good question.

If we're sending a log message that is just spaces, that's a bug. But we can't log from inside a logging function, unless we disable controller logs:
`
    ++disable_log_messages;
    tor_assert_nonfatal(*b);
    --disable_log_messages;
`

File: src/feature/control/control_events.c [hunk 2]
```diff
@@ -1318,6 +1318,16 @@ control_event_logmsg(int severity, log_domain_mask_t domain, const char *msg)
       for (cp = b; *cp; ++cp)
         if (*cp == '\r' || *cp == '\n')
           *cp = ' ';
+
+      /* Remove trailing spaces */
+      for (--cp; *cp == ' ' && cp >= b; --cp)
+        *cp = '\0';
+
+      if ( cp == b ){
```
Comment (teor2345) [inline src/feature/control/control_events.c]: The `cp == b` check isn't necessary, `tor_assert_nonfatal(*b);` makes sure that b is not an empty string.

**Pull request closed without merging**